pixeltable/utils/azure_store.py

```diff
@@ -0,0 +1,297 @@
+import logging
+import re
+import threading
+import uuid
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Iterator, NamedTuple, Optional
+
+from azure.core.exceptions import AzureError
+
+from pixeltable import env, exceptions as excs
+from pixeltable.config import Config
+from pixeltable.utils.object_stores import ObjectPath, ObjectStoreBase, StorageObjectAddress
+
+if TYPE_CHECKING:
+    from pixeltable.catalog import Column
+
+_logger = logging.getLogger('pixeltable')
+
+
+client_lock = threading.Lock()
+
+
+class AzureClientDict(NamedTuple):
+    """Container for actual Azure access objects (clients, resources)
+    Thread-safe, protected by the module lock 'client_lock'"""
+
+    profile: Optional[str]  # profile used to find credentials
```

Comment:
Are we not using '| None' in new code?

Reply:
It's not new code, it's adapted from Jim's existing PR. I'm planning to do a global search-and-replace to switch to `| None` but I'm waiting for the create_table() PR to merge. That'll be the easiest way to update these.